os/mm/mm_heap/mm_getheap.c

```diff
@@ -51,9 +51,88 @@ extern struct mm_heap_s g_mmheap[CONFIG_MM_NHEAPS];
  * Pre-processor Definitions
  ****************************************************************************/
 
+/****************************************************************************
+ * Public Varialbles
+ ****************************************************************************/
+#if defined(CONFIG_APP_BINARY_SEPARATION) && defined(__KERNEL__)
+typedef struct app_heap_s {
+	struct mm_heap_s *heap;
+	struct app_heap_s *next;
+	struct app_heap_s *prev;
+} app_heap_s;
+
+static app_heap_s *app_heap_list;
+extern struct mm_heap_s g_pheap;
+#endif
+
 /****************************************************************************
  * Public Functions
  ****************************************************************************/
+#if defined(CONFIG_APP_BINARY_SEPARATION) && defined(__KERNEL__)
+void mm_add_app_heap_list(struct mm_heap_s *heap)
+{
+	app_heap_s *node = (app_heap_s *)kmm_malloc(sizeof(app_heap_s));
+	node->heap = heap;
+
+	/* Add the new heap node to the head of the list*/
+	node->next = app_heap_list;
+	node->prev = NULL;
+	app_heap_list = node;
+}
+
+void mm_remove_app_heap_list(struct mm_heap_s *heap)
+{
+	app_heap_s *node;
+	/* Search the heap node in the list */
+	for (node = app_heap_list; node; node = node->next) {
+		if (node->heap == heap) {
+			/* If matching heap node is found.. */
+			if (node->prev) {
+				/* Heap node is in the middle of the list.
+				 * Adjust the prev pointer
+				 */
+				node->prev->next = node->next;
+			} else {
+				/* Heap node is at the head of the list.
+				 * Replace the list head with new value.
+				 */
+				app_heap_list = node->next;
+			}
+
+			if (node->next) {
+				/* Adjust the prev pointer of the next node
+				 * only if it is not the last node in the list.
+				 */
+				node->next->prev = node->prev;
+			}
+
+			/* Finally, we free the removed heap node */
+			kmm_free(node);
+		}
+	}
+}
+
+static struct mm_heap_s *mm_get_app_heap(void *address)
+{
+	app_heap_s *node;
+
+	/* First, search the address in list of app heaps */
+	for (node = app_heap_list; node; node = node->next) {
+		if ((address > (void *)node->heap->mm_heapstart[0]) && (address < (void *)node->heap->mm_heapend[0])) {
```

Comment:
Is it okay if heap has multiple region?

Reply:
No. It wont work for multiple regions. 
However, when we do APP separation, I am assuming that multiple regions are not present. 
It will be difficult to handle that scenario. 

I have already added below error message in partition manager.

```
#if CONFIG_MM_REGIONS > 1
#error "Unable to handle APP RAM PARTITIONS for CONFIG_MM_REGIONS > 1"
#endif

#if CONFIG_MM_NHEAPS > 1
#error "Unable to handle APP RAM PARTITIONS for CONFIG_MM_NHEAPS > 1"
#endif

```